deps/build.jl

```diff
@@ -1,52 +1,114 @@
 using BinDeps
+using BinDeps: MakeTargets
 
 basedir = dirname(@__FILE__)
 prefix = joinpath(basedir, "usr")
 
 cmake_version = v"3.7.2"
 base_url = "https://cmake.org/files/v$(cmake_version.major).$(cmake_version.minor)"
+@static if is_windows()
+    binary_name = "cmake.exe"
+else
+    binary_name = "cmake"
+end
+
+function install_binaries(file_base, file_ext, binary_dir)
+    filename = "$(file_base).$(file_ext)"
+    url = "$(base_url)/$(filename)"
+    binary_path = joinpath(basedir, "downloads", file_base, binary_dir)
+
+    @static if is_windows()
+        install_tool = cp
+    else
+        install_tool = symlink
+    end
+
+    install_step = () -> begin
+        for file in readdir(binary_path)
+            install_tool(joinpath(binary_path, file), 
+                         joinpath(prefix, "bin", file))
+        end
+    end
+
+    (@build_steps begin
+        FileRule(joinpath(prefix, "bin", binary_name), 
+            (@build_steps begin
+                FileDownloader(url, joinpath(basedir, "downloads", filename))
+                FileUnpacker(joinpath(basedir, "downloads", filename),
+                             joinpath(basedir, "downloads"), 
+                             "")
+                CreateDirectory(joinpath(prefix, "bin"))
+                install_step
+            end))
+    end)
+end
+
+function install_from_source(file_base, file_ext)
+    filename = "$(file_base).$(file_ext)"
+    url = "$(base_url)/$(filename)"
+
+    (@build_steps begin
+        FileRule(joinpath(prefix, "bin", binary_name), 
+            (@build_steps begin
+                FileDownloader(url, joinpath(basedir, "downloads", filename))
+                CreateDirectory(joinpath(basedir, "src"))
+                FileUnpacker(joinpath(basedir, "downloads", filename),
+                             joinpath(basedir, "src"), 
+                             "")
+                begin
+                    ChangeDirectory(joinpath(basedir, "src", file_base))
+                    `./configure --prefix=$(prefix)`
+                    MakeTargets()
+                    MakeTargets("install")
+                end
+            end))
+    end)
+end
+
+force_source_build = lowercase(get(ENV, "CMAKEWRAPPER_JL_BUILD_FROM_SOURCE", "")) in ["1", "true"]
 
-@static if is_linux()
-    download_name = "cmake-$(cmake_version)-Linux-x86_64"
-    filename = "$(download_name).tar.gz"
-    binary = "cmake"
-    binary_path = joinpath("bin", binary)
+process = @static if is_linux()
+    arch = strip(readstring(`arch`))
```

Comment:
are you sure this is always installed, even in minimal environments? why not use the Julia `Sys.ARCH` variable?